app/Http/Controllers/UserController.php

```diff
@@ -75,23 +84,176 @@ public function login(Request $request)
-        // ];
         $credentials = $request->only('email', 'password');
         try {
             // attempt to verify the credentials and create a token for the user
             if (! $token = JWTAuth::attempt($credentials)) {
-                return response()->json(['success' => false, 'error' => 'Invalid Credentials. Please make sure you entered the right information and you have verified your email address.'], 401);
+                return response()->json(
+                    ['success' => false,
+                    'error' => 'Invalid Credentials. Please make sure you entered a valid email address.'],
+                    401
+                );
             }
         } catch (JWTException $e) {
             // something went wrong whilst attempting to encode the token
             return response()->json(['success' => false, 'error' => 'could_not_create_token'], 500);
         }
-
         // all good so return the token
-        return response()->json(['success' => true, 'data'=> [ 'token' => $token ]]);
+        return response()->json(
+            [
+                'success' => true,
+                'data'=> [
+                    'token' => $token,
+                    'email' => $credentials['email'],
+                    ]
+            ]
+        );
+    }
+
+    /**
+     * Get current user
+     * @param Request $request
+     * @return \Illuminate\Http\JsonResponse
+     */
+
+    public function getUser(Request $request) 
+    {
+        if (!$user = JWTAuth::parseToken()->authenticate()) {
+            return response()->json(["message" => "User not found"], 404);
+        }
+    
+        $user_id = $user->id;
+        $user= User::where("id", "=", $user_id)->first();
+
+        return response()->json(['success' => true, 'user' => $user]);
+    }
+
+    /**
+     * Follow a user
+     * @param Request $request
+     * @return \Illuminate\Http\JsonResponse
+     */
+
+     public function followUser(Request $request, $userId)
+     {
+        if (!$user = JWTAuth::parseToken()->authenticate()) {
+            return response()->json(["message" => "User not found"], 404);
+        }
+
+        $user_id = $user->id;
+
+        if ($user_id == $userId) {
+            return response()->json([
+                "success" => false,
+                "message" => "You can not follow yourself"
+            ]);
+        }
+        
+        $user= User::where("id", "=", $userId)->first();
+
+        $follow = new Following();
+
+        $followed = $follow->where([
+            ['user_id', '=', $userId],
+            ['follower_id', '=', $user_id],
+        ])->first();
+
+        if (is_null($followed) == false) {
+            $follow->where([
+                ['user_id', '=', $userId],
+                ['follower_id', '=', $user_id],
+            ])->delete();
+            return response()->json([
+                'success' => false,
+                'current user unfollowed target user'], 200);
```

Comment:
You might want to provide a more descriptive error message. Please correct in other instances of the application